db.go

```diff
@@ -385,13 +407,18 @@ func (db *DB) checkpoint(ctx context.Context) error {
 	}
 
 	// Remove WAL file.
-	if err := os.Remove(db.WALPath()); err != nil {
-		return fmt.Errorf("remove wal: %w", err)
+	if err := db.TruncateWAL(ctx, 0); err != nil {
+		return fmt.Errorf("truncate wal: %w", err)
 	}
```

Comment:
Changed to a truncation as performing a delete while clients are active can cause issues.